tests/TestCase.php

```diff
@@ -2,9 +2,34 @@
 
 namespace Tests;
 
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\HasOne;
 use Illuminate\Foundation\Testing\TestCase as BaseTestCase;
 
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function testHasManyRelation($related, $foreignKey, $relation)
+    {
+        $this->assertInstanceOf(HasMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function testBelongToRelation($related, $foreignKey, $ownerKey, $relation)
+    {
+        $this->assertInstanceOf(BelongsTo::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($ownerKey, $relation->getOwnerKeyName());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
```

Comment:
giải thích các function

Reply:
Các function này em dùng để test relation của các model:
- $this->assertInstanceOf(HasMany::class, $relation); Kiểm tra xem có phải instance của HasMany
- $this->assertInstanceOf($related, $relation->getRelated()); Kiểm tra model có đúng với relationship 
- $this->assertEquals($foreignKey, $relation->getForeignKeyName()); Kiểm tra xem có đúng khóa ngoại
- $this->assertEquals($ownerKey, $relation->getOwnerKeyName()); Kiểm tra xem có đúng khóa chính